runtime/common/BaseRemoteSimulatorQPU.h

```diff
@@ -126,15 +126,38 @@ class BaseRemoteSimulatorQPU : public cudaq::QPU {
     // set up a single-shot execution context for this case.
     static thread_local cudaq::ExecutionContext defaultContext("sample",
                                                                /*shots=*/1);
+    // This is a kernel invocation outside the CUDA-Q APIs (sample/observe).
+    const bool isDirectInvocation = executionContextPtr == nullptr;
```

Comment:
No worries here.
```suggestion
    const bool isDirectInvocation = !executionContextPtr;
```
I think way back we decided to not write out comparisons like `== 0` and `!= 0` in the code. I'm sure we've meandered off that style in many places. At any rate, we may want to discuss again.